707.py

```diff
@@ -0,0 +1,91 @@
+class Node:
+
+    def __init__(self, val: int):
+        self.val = val
+        self.next = None
+
+class MyLinkedList:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.pre_head = Node(0)
```

Comment:
This is an annoying problem, but as long as you have this "dummy" `pre_head` node, you have already avoided (probably) 90% of the bugs related to linked list problems.